qb/test/select.test.ts

```diff
@@ -317,3 +332,101 @@ test("nonchainable offset/limit", () => {
 
   const f3: val2 extends $expr_Select ? true : false = true;
 });
+
+test("fetch heroes", async () => {
+  const result = await pool.query(e.select(e.Hero).toEdgeQL());
+  expect(result.length).toEqual(3);
+  expect(result.every((h) => typeof h.id === "string")).toEqual(true);
+});
+test("filter by id", async () => {
+  const result = await e
+    .select(e.Hero)
+    .filter(e.eq(e.Hero.id, e.uuid(data.spidey.id)))
+    .query(pool);
+
+  expect(result?.id).toEqual(data.spidey.id);
+});
+test("limit 1", async () => {
+  const query = e.select(e.Hero).orderBy(e.Hero.name).offset(1).limit(1);
+  const result = await query.query(pool);
+  expect(result?.id).toEqual(data.iron_man.id);
+});
+
+test("limit 2", async () => {
+  const query = e.select(e.Hero).orderBy(e.Hero.name).offset(1).limit(2);
+  const results = await query.query(pool);
+
+  expect(results.length).toEqual(2);
+  expect(results).toEqual([{id: data.iron_man.id}, {id: data.spidey.id}]);
+});
+
+test("shapes", async () => {
+  const query = e
+    .select(e.Hero, {
+      id: true,
+      name: true,
+      secret_identity: true,
+      villains: {id: true},
+    })
+    .filter(e.eq(e.Hero.name, e.str("Iron Man")));
```

Comment:
Do we have assertSingle implemented?

I'm asking because if we do, then that would make it a better place, but also you could then

`expect(result).toMatchObject(data.iron_man)`

Reply:
Good call, updated.